### ISSUE TITLE
Recap Search Returning Non-Recap Documents

### ISSUE THREAD

**Issue by flooie** [freelawproject/courtlistener#3702]
###  Description
Recap search is currently returning documents that are not strictly Recap documents. This issue seems to have originated last year when all documents were indexed in Recap, including those not inherently belonging to Recap. Since the switch over to the frontend using Elasticsearch, these documents have started to surface with incorrect or bad data.  

### Expected Behavior
Recap search should only return documents that are genuine Recap documents, ensuring data integrity and relevance in search results.

### Current Behavior
Non-Recap documents are being returned in the search results. This includes documents with data not labeled because it assumes recap documents have date filed values in the docket.  Unfortunately this appears to make the frontend look like bad data, which can mislead users and degrade the quality of the search functionality.  See following query and screen shot.

https://www.courtlistener.com/?q=&type=r&order_by=entry_date_filed%20desc&case_name=Roe%20v.%20Wade
<img width="935" alt="Screenshot 2024-01-31 at 2 44 23 PM" src="https://github.com/freelawproject/courtlistener/assets/6464529/e751d95c-8251-4345-ad47-f796c485909a">

### Steps to Reproduce

Perform a search using the Recap search for any opinion you know of in the opinion db.  It should appear without a date filed.

### Possible Solution

A review of the indexing process implemented last year might be necessary to filter out non-Recap documents. Additionally, ensuring that the Elasticsearch frontend aligns correctly with the intended data set may resolve this issue.

### Additional Context
What's interesting is that it's almost halfway to what we want to complete the other way around.  If we can reverse this issue - we could pretty nicely unify the opinion search and the opinions in recap search together.  

@mlissner @albertisfu

**Comment by albertisfu** [comment]
Yeah the root issue here is that in Solr search, indexing was based on `RECAPDocuments` instead of `Dockets` and `RECAPDocuments`, as we do now in ES.

Previously, in Solr, it was not possible to index a Docket that doesn't belong to the RECAP collection because all the filings came from RECAP. So, if a Docket added from an Opinion received entries, it was indexed and marked RECAP as its source.

Now, we're indexing every docket in the database; this includes dockets without filings from Opinions.

- The solution, as Bill mentioned, seems to be to exclude dockets from indexing where RECAP is not part of the source.
- Currently, `source` is not an indexed field in ES DocketDocument. Might it be useful to add it?
- We'll need to do a cleanup of these dockets. We can create a command that filters out all the dockets where `RECAP` is not part of its `source` and remove them from the ES index.

**Comment by mlissner** [comment]
Yeah, I'm definitely aware of this one, and I've been on the fence about what to do about it. Right now the results are a bit messy, but is it a bad thing that RECAP search returns case law results too? I guess it's not what people are expecting and we should probably remove it, lest it cause more confusion. 

I'll get this prioritized on your backlog, Alberto.

**Comment by mlissner** [comment]
@albertisfu should we do this before we start our big reindex? I imagine having something like 7M fewer items in the index would be good, yes?

**Comment by albertisfu** [comment]
Yes, definitely. It makes sense to do this now to avoid adding additional content that is not from RECAP. I see this as part of the [sweep indexer ](https://github.com/freelawproject/courtlistener/pull/3862) so it filters out dockets where RECAP is not part of their source.

Also, as an additional step, either in the `sweep_indexer` or in `cl_index_parent_and_child_docs` (I think it makes more sense here since it will be a one-time run), it should filter out dockets that are not part of the RECAP collection and remove them from the index.

**Comment by mlissner** [comment]
Wouldn't it be better just to do that as a separate little script so we can throw it away later?

**Comment by albertisfu** [comment]
Got it. Yes, the second step of removing all the cases that don't belong to RECAP makes sense for creating a separate script.

The first step of avoiding adding content that doesn't belong to RECAP should be part of the `sweep_indexer`, and the ES indexing signals would also need this tweak.

**Comment by mlissner** [comment]
Perfect. Yeah, let's do that then before our do our sweep at long last! OTOH, could we start the sweep as soon as it's ready and just do this/run this once this is ready? That'd get the job done faster, right?

**Comment by albertisfu** [comment]
Sure. The sweep indexer is ready with the tweak to avoid indexing Dockets that don't belong to RECAP, and I've opened an issue to remind us to apply the same tweak to indexing signals.

As part of this issue, we'll also develop the script to remove Dockets that are not part of RECAP.
